prime-router/src/main/kotlin/fhirengine/engine/FHIRConverter.kt

```diff
@@ -77,16 +104,15 @@ class FHIRConverter(
         actionLogger: ActionLogger,
         actionHistory: ActionHistory,
     ): List<FHIREngineRunResult> {
+        actionLogger.setReportId(queueMessage.reportId)
+        actionHistory.trackExistingInputReport(queueMessage.reportId)
         val format = Report.getFormatFromBlobURL(queueMessage.blobURL)
         logger.trace("Processing $format data for FHIR conversion.")
-        val fhirBundles = when (format) {
-            Report.Format.HL7, Report.Format.HL7_BATCH -> getContentFromHL7(queueMessage, actionLogger)
-            Report.Format.FHIR -> getContentFromFHIR(queueMessage, actionLogger)
-            else -> throw NotImplementedError("Invalid format $format ")
-        }
+
+        val fhirBundles = process(format, queueMessage, actionLogger)
+
+        // TODO add more logging here
```

Comment:
reminder. would a `withLoggingContext(contextMap) {` (like in FHIRTranslator.kt) be applicable for convert function?

Reply:
Yep, though I'm going to defer that to a follow up